README.md

```diff
@@ -1,20 +1,88 @@
+
+## Architecture
+
+### Components
+
+The Flow Data Provisioning Service is constituted of three main components.
+
+The **indexer** is at the core of the service. It indexes deltas into its own data structures and provides a low level
+API to expose its state.
+The **mapper** synchronizes the **indexer** with a source of deltas (`FilesystemChain` or `NetworkChain`.)
+The **chain** is in charge of fetching block sequence data (from a WAL in the filesystem for past sporks, or a network
+socket for live ones.)
 
-- index transaction events
-- events retrieval interface
-- REST API interface
-- DPS web client
+### Diagram
 
-## Interfaces
+The following diagram describes a simple overview of an example where the DPS is reading from two sporks. One of them
+is a live one, which sends
 
-1. Indexer: indexes changes/deltas per block
-2. Update Stream: gets all the trie updates per chunk
-3. Chain View: has block sequence data
\ No newline at end of file
+```text
+┌─────────────────┐
+│                 │
+│   Past Spork    │
+│                 │
+│  ┌───────────┐  │             ┌───────┐
+│  │           │  │             │       │
+│  │ Exec Node ├──┼────────────►│ WALog │
+│  │           │  │             │       │
+│  └───────────┘  │             └───┬───┘
+│                 │                 │
+│                 │                 │
+└─────────────────┘                 │
+                                    ▼
+                      ┌─────────────────────────────┐
+                      │  FSChain / LedgerWALMapper  │
+                      ├────────────▼▼▼──────────────┤
+                      │                             │                          ┌──────────┐
+                      │                             │  `GET /account/balance`  │          │
+                      │  DATA PROVISIONING SERVICE  │◄─────────────────────────┤  Client  │
+                      │          INDEXER            │                          │          │
+                      │                             │                          └──────────┘
+                      │                             │
+                      ├────────────▲▲▲──────────────┤
+                      │  NetworkChain / LiveMapper  │
+                      └─────────────────────────────┘
+                                    ▲
+                                    │
+┌─────────────────┐                 │
+│                 │                 │
+│  Live Network   │                 │
+│                 │                 │
+│  ┌───────────┐  │                 │
+│  │           │  │  `POST /block`  │
+│  │ Exec Node ├──┼─────────────────┘
+│  │           │  │
+│  └───────────┘  │
+│                 │
+│                 │
+└─────────────────┘
```

Comment:
On this part, there is indeed an execution node, with a `publish` socket. I'm not really sure what the `POST /block` is about; there are two updates pushed by the execution node: `trie updates` and `transaction events`. Both would feed into the `Live Mapper`.

The missing part here is that there is also an `access node`, and the `NetworkChain` would request `block headers` from its GRPC API.